protocol/contracts/templegold/TempleTeleporter.sol

```diff
@@ -0,0 +1,110 @@
+     */
+    function teleport(
+        uint32 dstEid,
+        address to,
+        uint256 amount,
+        bytes calldata options
+    ) external payable override returns(MessagingReceipt memory receipt) {
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        // Encodes the message before invoking _lzSend.
+        bytes memory _payload = abi.encodePacked(to.addressToBytes32(), amount);
+        // debit
+        temple.burnFrom(msg.sender, amount);
+        emit TempleTeleported(dstEid, msg.sender, to, amount);
+
+        receipt = _lzSend(dstEid, _payload, options, MessagingFee(msg.value, 0), payable(msg.sender));
+    }
+
+    /**
+     * @dev External function to interact with the LayerZero EndpointV2.quote() for fee calculation.
+     * @param _dstEid The destination endpoint ID.
+     * @param _message The message payload.
+     * @param _options Additional options for the message.
+     * @return fee The calculated MessagingFee for the message.
+     *      - nativeFee: The native fee for the message.
+     *      - lzTokenFee: The LZ token fee for the message.
+     */
+    function quote(
+        uint32 _dstEid,
+        bytes memory _message,
+        bytes memory _options
+    ) external view returns (MessagingFee memory fee) {
+        return _quote(_dstEid, _message, _options, false);
+    }
+
+    /**
+     * @dev External function to interact with the LayerZero EndpointV2.quote() for fee calculation.
+     * @param _dstEid The destination endpoint ID.
+     * @param _to Recipient
+     * @param _amount Amount to send
+     * @param _options Additional options for the message.
+     * @return fee The calculated MessagingFee for the message.
+     *      - nativeFee: The native fee for the message.
+     *      - lzTokenFee: The LZ token fee for the message.
+     */
+    function quote(
+        uint32 _dstEid,
+        address _to,
+        uint256 _amount,
+        bytes memory _options
+    ) external view returns (MessagingFee memory fee) {
+        return _quote(_dstEid, abi.encodePacked(_to.addressToBytes32(), _amount), _options, false);
+    }
+
+    /// @dev Called when data is received from the protocol. It overrides the equivalent function in the parent contract.
+    /// Protocol messages are defined as packets, comprised of the following parameters.
+    /// @param _payload Encoded message.
+    function _lzReceive(
+        Origin calldata /*_origin*/,
+        bytes32 /*_guid*/,
+        bytes calldata _payload,
+        address /*_executor,*/,  // Executor address as specified by the OApp.
+        bytes calldata /*_extraData */ // Any extra data or options to trigger on receipt.
+    ) internal override {
+        // Decode the payload to get the message
+        (address _recipient, uint256 _amount) = abi.decode(_payload, (address, uint256));
+        temple.mint(_recipient, _amount);
```

Comment:
Nasty scenario:

Apps in both chains are wired up correctly, but we forget to set the Teleporter as a valid minter of Temple token. 
The `teleport()` function will burn the tokens, send the LZ message, and in the receiving chain, `tempe.mint()` will revert, effectively losing tokens for the users.

I don't think there is an easy solution besides remembering to set Teleporter as a valid minter. So this is only a friendly reminder of how critical would be not to do it.

Reply:
Very good point, thanks for pointing out